packages/react-native-renderer/src/ReactNativeAttributePayloadFabric.js

```diff
@@ -287,14 +287,23 @@ function diffProperties(
     prevProp = prevProps[propKey];
     nextProp = nextProps[propKey];
 
-    // functions are converted to booleans as markers that the associated
-    // events should be sent from native.
     if (typeof nextProp === 'function') {
-      nextProp = (true: any);
-      // If nextProp is not a function, then don't bother changing prevProp
-      // since nextProp will win and go into the updatePayload regardless.
-      if (typeof prevProp === 'function') {
-        prevProp = (true: any);
+      if (typeof attributeConfig === 'object') {
+        // When the config for a function prop has a custom process method
+        // we don't assume its a regular event handler, but use the process method:
+        nextProp = attributeConfig.process(nextProp);
+        if (typeof prevProp === 'function') {
+          prevProp = attributeConfig.process(prevProp);
```

Comment:
It doesn't seem right that we process here before diffing - for other properties we first diff and then we process. Assuming that we have we have a `process` method here also doesn't match the strict validation we do in the rest of the method.

This whole method seems to invoke `process` at multiple different points though, so it seem hard to keep the default for functions.

Could you try something like this?

```
const attributeConfigHasProcess = typeof attributeConfig === 'object' && typeof attributeConfig.process === 'function';
if (typeof nextProp === 'function' && !attributeConfigHasProcess) {
  // keep as before
}

/* rely on process being invoked as it does for other methods */

Reply:
yeah I see, that makes much more sense!